paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethodKtx.kt

```diff
@@ -29,65 +31,89 @@ internal fun SupportedPaymentMethod.getPMAddForm(
 ) = requireNotNull(getSpecWithFullfilledRequirements(stripeIntent, config))
 
 /**
- * This function will determine if there is a valid for the payment method
+ * This function will determine if there is a valid spec for the payment method
  * given the [PaymentSheet.Configuration] and the [StripeIntent]
  */
 internal fun SupportedPaymentMethod.getSpecWithFullfilledRequirements(
     stripeIntent: StripeIntent,
     config: PaymentSheet.Configuration?
 ): LayoutFormDescriptor? {
-    val formSpec = formSpec
+    if (!stripeIntent.paymentMethodTypes.contains(code)) {
+        return null
+    }
+
+    val containsValidShippingInfo = when (stripeIntent) {
+        is PaymentIntent -> stripeIntent.containsValidShippingInfo
+        is SetupIntent -> false
+        is DeferredIntent -> config?.shippingDetails?.containsValidShippingInfo == true
+    }
+
+    val isPaymentFlow = stripeIntent is PaymentIntent ||
+        (stripeIntent as? DeferredIntent)?.mode is DeferredIntent.Mode.Payment
+
+    return getLayoutFormDescription(
+        isPaymentFlow,
+        this,
+        stripeIntent,
+        supportsPaymentIntentSfuSet(containsValidShippingInfo, config),
+        supportsPaymentIntentSfuSettable(containsValidShippingInfo, config),
+        supportsPaymentIntentSfuNotSettable(containsValidShippingInfo, config),
+        supportsSetupIntent(config)
```

Comment:
Any reason for moving the computations here? I feel like they can stay in the `when` statement in `getLayoutFormDescription()`.